services/app-api/src/resolvers/contract/indexContracts.ts

```diff
@@ -0,0 +1,145 @@
+    isCMSUser,
+    isAdminUser,
+    isBusinessOwnerUser,
+} from '../../domain-models'
+import type { ContractType } from '../../domain-models'
+import { isHelpdeskUser } from '../../domain-models/user'
+import type { QueryResolvers } from '../../gen/gqlServer'
+import { logError, logSuccess } from '../../logger'
+import type { Store } from '../../postgres'
+import { NotFoundError } from '../../postgres'
+import {
+    setErrorAttributesOnActiveSpan,
+    setResolverDetailsOnActiveSpan,
+    setSuccessAttributesOnActiveSpan,
+} from '../attributeHelper'
+import { GraphQLError } from 'graphql/index'
+import type { ContractOrErrorArrayType } from '../../postgres/contractAndRates/findAllContractsWithHistoryByState'
+
+const parseContracts = (
+    contractsWithHistory: ContractOrErrorArrayType,
+    span?: Span
+): ContractType[] => {
+    // separate valid contracts and errors
+    const parsedContracts: ContractType[] = []
+    const errorParseContracts: string[] = []
+    contractsWithHistory.forEach((parsed) => {
+        if (parsed.contract instanceof Error) {
+            errorParseContracts.push(
+                `${parsed.contractID}: ${parsed.contract.message}`
+            )
+        } else {
+            parsedContracts.push(parsed.contract)
+        }
+    })
+
+    // log all contracts that failed parsing to otel.
+    if (errorParseContracts.length > 0) {
+        const errMessage = `Failed to parse the following contracts:\n${errorParseContracts.join(
+            '\n'
+        )}`
+        logError('indexHealthPlanPackagesResolver', errMessage)
+        setErrorAttributesOnActiveSpan(errMessage, span)
+    }
+
+    return parsedContracts
+}
+
+const validateAndReturnContracts = (results: ContractType[], span?: Span) => {
+    const contracts: ContractType[] = results
+
+    const edges = contracts.map((contract) => {
+        return {
+            node: {
+                ...contract,
+            },
+        }
+    })
+
+    logSuccess('indexContracts')
+    setSuccessAttributesOnActiveSpan(span)
+    return { totalCount: edges.length, edges }
+}
+
+export function indexContractsResolver(
+    store: Store
+): QueryResolvers['indexContracts'] {
+    return async (_parent, _args, context) => {
+        const { user, ctx, tracer } = context
+        const span = tracer?.startSpan('indexContracts', {}, ctx)
+        setResolverDetailsOnActiveSpan('indexContracts', user, span)
+
+        if (isStateUser(user)) {
+            const contractsWithHistory =
+                await store.findAllContractsWithHistoryByState(user.stateCode)
+
+            if (contractsWithHistory instanceof Error) {
+                const errMessage = `Issue finding contracts with history by stateCode: ${user.stateCode}. Message: ${contractsWithHistory.message}`
+                logError('indexContracts', errMessage)
+                setErrorAttributesOnActiveSpan(errMessage, span)
+
+                if (contractsWithHistory instanceof NotFoundError) {
```

Comment:
can we return NotFoundError from findAllContracts*? I assume we just return empty array if there's nothing there

Reply:
Yes if the contract is not an id found in the db wouldnt it would be a NotFoundError

Reply:
@macrael @haworku what's the requested change here? findAllContractsWithHistoryByState can return either ContractOrErrorArrayType | NotFoundError | Error

Reply:
No change! I forgot it could actually return NotFoundError